pkg/controller/pravega/pravega_segmentstore.go

```diff
@@ -151,44 +141,48 @@ func makeSegmentstorePodSpec(pravegaCluster *api.PravegaCluster) corev1.PodSpec
 	return podSpec
 }
 
-func MakeSegmentstoreConfigMap(pravegaCluster *api.PravegaCluster) *corev1.ConfigMap {
+func MakeSegmentstoreConfigMap(p *api.PravegaCluster) *corev1.ConfigMap {
 	javaOpts := []string{
-		"-Xms1g -Xmx4g -XX:MaxDirectMemorySize=1g -Dpravegaservice.clusterName=" + pravegaCluster.Name,
+		"-Xms1g",
+		"-XX:+UnlockExperimentalVMOptions",
```

Comment:
Are these options configurable as part of another PR, right?

Reply:
No, JVM options are not externally configurable.

Reply:
Ok, this may not be great, because we will rely on the default `-Xmx` value set by the JVM implementation. The default may be set as follows:
> Often its default value is 1/4th of your physical memory or 1GB (whichever is smaller).

So, we may end up providing 4GBs of memory to a pod, but 1GB at most will be devoted to the JVM (despite the main purpose of that pod is to run the JVM for the Pravega process at hand). What is you opinion about this? Have you used `-XX:+PrintFlagsFinal` to see what is the actual `-Xmx` value that is being used? Would it make sense to set the `-Xmx` value in function to the provisioned memory resources for the pod?